python/mxnet/kvstore.py

```diff
@@ -10,31 +10,39 @@
 from .base import NDArrayHandle, KVStoreHandle
 from . import optimizer as opt
 
-def _ctype_key_value(keys, vals):
-    """
-    Returns ctype arrays for the key-value args. For internal use.
-    """
-    if isinstance(keys, int):
+def _ctype_str_key_value(keys, vals):
+    names = []
+    if isinstance(keys, str):
         if isinstance(vals, NDArray):
-            return (c_array(ctypes.c_int, [keys]),
+            names.append(c_str(keys))
```

Comment:
no need for _cast_to_str_keys

always cast to str directly.
names.append(c_str(str(keys)))